src/auth/baseexternalclient.ts

```diff
@@ -316,6 +338,9 @@ export abstract class BaseExternalAccountClient extends AuthClient {
       });
       this.projectId = response.data.projectId;
       return this.projectId;
+    } else if (!this.clientAuth && this.workforcePoolUserProject) {
+      // Return workforcePoolUserProject if client auth is not defined.
+      return this.workforcePoolUserProject;
```

Comment:
This won't work. The `workforce_pool_user_project` value can be a project ID or number. In fact, right now, it is only a project number. We have to call cloud resource manager. Please see the Python PR.

You can fix by doing something like:

```
    const projectNumber = this.projectNumber || this.workforcePoolUserProject;
    if (this.projectId) {
      // Return previously determined project ID.
      return this.projectId;
    } else if (projectNumber) {
      // Preferable not to use request() to avoid retrial policies.
      const headers = await this.getRequestHeaders();
      const response = await this.transporter.request<ProjectInfo>({
        headers,
        url: `${CLOUD_RESOURCE_MANAGER}${projectNumber}`,
        responseType: 'json',
      });
      this.projectId = response.data.projectId;
      return this.projectId;
    } ...
```